libraries/chain/include/eosio/chain/trace.hpp

```diff
@@ -45,6 +45,43 @@ namespace eosio { namespace chain {
       std::optional<fc::exception>    except;
       std::optional<uint64_t>         error_code;
       std::vector<char>               return_value;
+
+      digest_type digest_savanna() const {
+         assert(!!receipt);
+         const action_receipt& r = *receipt;
+
+         digest_type::encoder e;
+         fc::raw::pack(e, r.receiver);
+         fc::raw::pack(e, r.recv_sequence);
+         fc::raw::pack(e, act.account);
+         fc::raw::pack(e, act.name);
+         fc::raw::pack(e, r.act_digest);
+
+         {
```

Comment:
Why is this nested encoder needed?

Reply:
Because of this in the issue comment:
<img width="692" alt="image" src="https://github.com/AntelopeIO/leap/assets/1496907/ac814867-f1da-4905-8bdf-16f79ff89eb7">